src/policy/marksweepspace/native_ms/block.rs

```diff
@@ -285,9 +285,12 @@ impl Block {
         let mut cell = self.start();
         let mut last = unsafe { Address::zero() };
         while cell + cell_size <= self.start() + Block::BYTES {
+            #[cfg(not(feature = "extra_header"))]
             // The invariants we checked earlier ensures that we can use cell and object reference interchangably
             // We may not really have an object in this cell, but if we do, this object reference is correct.
             let potential_object = ObjectReference::from_raw_address(cell);
+            #[cfg(feature = "extra_header")]
+            let potential_object = ObjectReference::from_raw_address(cell + VM::EXTRA_HEADER_BYTES);
```

Comment:
The binding can config `EXTRA_HEADER_BYTES`, and there is no requirement on what value a binding can use. If  a binding uses a value that is not a multiple of the fixed alignment, this code is wrong. You can either introduce some requirements for a 'valid EXTRA_HEADER_BYTES value', or simply change the condition in L264 so we use `native_brute_force_sweep` if `extra_header` is enabled.